sparta/sparta/events/PhasedPayloadEvent.hpp

```diff
@@ -127,13 +139,14 @@ namespace sparta
                             << parent_->name_ << " to handler: "
                             << target_consumer_event_handler_.getName());
                 scheduled_ = false;
-                target_consumer_event_handler_((const void*)&payload_);
+                target_consumer_event_handler_((const void*)payload_);
                 reclaim_();
             }
 
             PhasedPayloadEvent<DataT> * parent_ = nullptr;
             const SpartaHandler         target_consumer_event_handler_;
-            DataT                       payload_;
+            DataT *                     payload_;
+            std::byte                   payload_storage_[sizeof(DataT)];
```

Comment:
You might need to align this memory just to be safe.  See the example: https://en.cppreference.com/w/cpp/memory/construct_at.html